### PR TITLE
SASS-2507 create unauthorised payments view model

### PR DESCRIPTION
**Pull request by ILee15**
### Description
Create unauthorised payments view model
Changed redirect on Uk Pension Scheme Payments page to render the page (as it is the first page in the journey) and prevent redirecting to CYA page if no session data which causes it to loop forever

Add a link to the relevant story in Jira
[SASS-2507](https://jira.tools.tax.service.gov.uk/browse/SASS-2507)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [ ]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by erasmos-hmrc** [comment]
This is unrelated to these changes: I'm surprised that Mongo needs to be running in order to run the unit tests.

I would have thought that would only be for integration tests.

File: it/utils/PensionDataStubs.scala [hunk 1]
```diff
@@ -257,7 +258,7 @@ object PensionUserDataStub extends IntegrationTest {
                        taxyear: Int = taxYear,
                        isPriorSubmission: Boolean = true,
                        cya: PensionsCYAModel = (PensionsCYAModel(paymentsIntoPensionViewModel, pensionsAnnualAllowancesViewModel,
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/utils/PensionDataStubs.scala:260]: There are two unused imports:
```
import utils.PensionUserDataStub.pensionsAnnualAllowancesViewModel
import utils.IntegrationTest
```

File: it/repositories/PensionsUserDataRepositoryISpec.scala [hunk 2]
```diff
@@ -64,8 +64,9 @@ class PensionsUserDataRepositoryISpec extends IntegrationTest with FutureAwaits
   private val pensionAnnualAllowances = PensionAnnualAllowancesViewModel(reducedAnnualAllowanceQuestion = Some(true))
   private val pensionLifetimeAllowances = PensionLifetimeAllowancesViewModel(aboveLifetimeAllowanceQuestion = Some(true))
   private val incomeFromPensions = IncomeFromPensionsViewModel(statePension = Some(anStateBenefitViewModelOne))
+  private val unauthorisedPayments = UnauthorisedPaymentsViewModel(unauthorisedPaymentsQuestion = Some(true))
 
-  private val pensionCYAModel = PensionsCYAModel(paymentsIntoPension, pensionAnnualAllowances, pensionLifetimeAllowances, incomeFromPensions)
+  private val pensionCYAModel = PensionsCYAModel(paymentsIntoPension, pensionAnnualAllowances, pensionLifetimeAllowances, incomeFromPensions, unauthorisedPayments)
```
Comment (erasmos-hmrc) [inline it/repositories/PensionsUserDataRepositoryISpec.scala:69]: Could we make this line shorter?

File: it/controllers/pensions/paymentsIntoPensions/RetirementAnnuityControllerISpec.scala [hunk 3]
```diff
@@ -23,6 +23,7 @@ import builders.PensionAnnualAllowanceViewModelBuilder.aPensionAnnualAllowanceEm
 import builders.PensionLifetimeAllowanceViewModelBuilder.aPensionLifetimeAllowancesEmptyViewModel
 import builders.PensionsCYAModelBuilder._
 import builders.PensionsUserDataBuilder
+import builders.PensionsUserDataBuilder.pensionsUserDataWithPaymentsIntoPensions
```
Comment (erasmos-hmrc) [inline it/controllers/pensions/paymentsIntoPensions/RetirementAnnuityControllerISpec.scala:26]: A few obsolete imports.

File: app/services/PensionSessionService.scala [hunk 4]
```diff
@@ -168,10 +168,34 @@ class PensionSessionService @Inject()(pensionUserDataRepository: PensionsUserDat
         //TODO: set the question below based on the list from backend
         uKPensionIncomesQuestion = Some(getUkPensionIncome(prior).nonEmpty),
         uKPensionIncomes = getUkPensionIncome(prior)
+      ),
+
+      unauthorisedPayments = UnauthorisedPaymentsViewModel(
+        unauthorisedPaymentsQuestion = getUnauthorisedPaymentsQuestion(prior),
```
Comment (erasmos-hmrc) [inline app/services/PensionSessionService.scala:174]: I like how you're using a function here; much more concise than how we do it below.

Perhaps which should do the same with those, sometime?

File: app/services/EncryptionService.scala [hunk 5]
```diff
@@ -120,7 +121,8 @@ class EncryptionService @Inject()(secureGCMCipher: SecureGCMCipher, appConfig: A
       paymentsIntoPension = decryptPaymentsIntoPensionViewModel(pension.encryptedPaymentsIntoPension),
       pensionsAnnualAllowances = decryptPensionAnnualAllowanceViewModel(pension.encryptedPensionAnnualAllowances),
       pensionLifetimeAllowances = pension.pensionLifetimeAllowances.decrypted(),
-      incomeFromPensions = pension.incomeFromPensions.decrypted()
+      incomeFromPensions = pension.incomeFromPensions.decrypted(),
+      unauthorisedPayments = pension.unauthorisedPayments.decrypted()
```
Comment (erasmos-hmrc) [inline app/services/EncryptionService.scala:125]: Could this be moved into `PensionsCYAModel` now (both `decryptPensions` and `encryptPension`)? 

Or should we wait until [SASS-3049](https://jira.tools.tax.service.gov.uk/browse/SASS-3049)?

(I may have misunderstood)

**Review by erasmos-hmrc — DISMISSED** [review]
Note that I successfully ran the unit and integration tests, but not those for the journey, as I don't yet know how.

File: app/controllers/pensions/incomeFromPensions/UkPensionSchemePaymentsController.scala [hunk 6]
```diff
@@ -58,7 +58,7 @@ class UkPensionSchemePaymentsController @Inject()(implicit val mcc: MessagesCont
             case _ => Future.successful(Ok(view(yesNoForm(request.user), taxYear)))
           }
         case _ =>
-          Future.successful(Redirect(UkPensionIncomeCYAController.show(taxYear)))
+          Future.successful(Ok(view(yesNoForm(request.user), taxYear)))
```
[the remark this answers is not in the document]
Comment (ILee15) [reply app/controllers/pensions/incomeFromPensions/UkPensionSchemePaymentsController.scala:61]: Yeah I can add a note to the story. It's the first discovery so no ticket for it

**Review by HarryGull — APPROVED** [review]

**Pull request merged**